modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheProcessor.java

```diff
@@ -2839,8 +2852,11 @@ private void processCacheStopRequestOnExchangeDone(ExchangeActions exchActions)
             throw new IgniteException(msg, e);
         }
 
-        for (IgniteBiTuple<CacheGroupContext, Boolean> grp : grpsToStop)
-            stopCacheGroup(grp.get1().groupId(), grp.get2());
+        for (IgniteBiTuple<CacheGroupContext, Boolean> grp : grpsToStop) {
+            Boolean stat = grpMetricsToStop.get(grp.get1().groupId());
```

Comment:
Are we really need  intermediate `grpMetricsToStop` map?
Let's just calculate `statisticsEnabled` flag right in place.

Reply:
yep ... cause upper we removes it context : prepareCacheStop -> sharedCtx.removeCacheContext(ctx);